tests/testthat/test-translate-package.R

```diff
@@ -19,543 +19,430 @@ test_that("translate_package arg checking errors work", {
-    {
-      expect_messages(
-        translate_package(pkg, verbose=TRUE),
-        c("Generating .pot files", "No languages provided"),
-        fixed = TRUE
-      )
-
-      pkg_files <- list.files(pkg, recursive=TRUE)
-
-      pot_file <- "po/R-rMsg.pot"
-      expect_true(pot_file %in% pkg_files)
-      # testing gettextf's ... arguments are skipped
-      expect_all_match(readLines(file.path(pkg, pot_file)), "don't translate me", invert=TRUE, fixed=TRUE)
-
-      # Non-UTF-8 machines don't run en@quot translations by default.
-      #   Mostly applies to Windows, but can also apply to Unix
-      #   (e.g. r-devel-linux-x86_64-debian-clang on CRAN), #191
-      if (l10n_info()[["UTF-8"]]) {
-        expect_match(pkg_files, "inst/po/en@quot/LC_MESSAGES/R-rMsg.mo", all = FALSE)
-      }
-    }
-  )
-  # do translations with mocked input
-  prompts <- restore_package(
-    pkg,
-    tmp_conn = mock_translation("test-translate-package-r_msg-1.input"),
-    {
-      expect_messages(
-        translate_package(pkg, "zh_CN", verbose=TRUE),
-        c("Beginning new translations", "BEGINNING TRANSLATION", '"Installing" translations with msgfmt'),
-        fixed = TRUE
-      )
-
-      pkg_files <- list.files(pkg, recursive = TRUE)
-
-      expect_true("po/R-zh_CN.po" %in% pkg_files)
-      expect_match(pkg_files, "inst/po/zh_CN/LC_MESSAGES/R-rMsg.mo", all = FALSE, fixed = TRUE)
-
-      zh_translations <- readLines(file.path(pkg, "po/R-zh_CN.po"), encoding='UTF-8')
-
-      expect_match(zh_translations, "Last-Translator.*test-user.*test-user@github.com", all = FALSE)
-      expect_match(zh_translations, "早上好", all = FALSE)
-      # plural message
-      expect_match(zh_translations, "该起床了", all = FALSE)
-    }
-  )
-  expect_all_match(prompts, c("^---^", "^^"), fixed=TRUE)
-
-  # all translations already done
-  restore_package(
-    pkg,
-    {
-      expect_messages(
-        translate_package(pkg, "fa", verbose=TRUE),
-        "Translations for fa are up to date! Skipping",
-        fixed = TRUE
-      )
-    }
-  )
+with_restoration_test_that("translate_package handles a data package (no R dir)", "r_data_pkg", {
+  expect_snapshot(translate_package(verbose=TRUE))
 })
 
-test_that("translate_package works on package with outdated (fuzzy) translations", {
-  # simple run-through without doing translations
-  prompts = restore_package(
-    pkg <- test_package("r_fuzzy"),
-    tmp_conn = mock_translation("test-translate-package-r_fuzzy-1.input"),
-    {
-      expect_messages(
-        translate_package(pkg, "zh_CN", verbose=TRUE),
-        c("translations marked as deprecated", "SINGULAR MESSAGES", "PLURAL MESSAGES"),
-        fixed = TRUE
-      )
-    }
-  )
-  expect_match(prompts, "a similar message was previously translated as", all=FALSE)
-})
+with_restoration_test_that("translate_package works on a simple package w/o translating", "r_msg", {
+  mockery::stub(translate_package, "get_atime", "0000-01-01 00:00:00")
```

Comment:
@hadley I couldn't seem to make `mockery::stub` work inside `with_restoration_test_that`, are you familiar enough with whether that's possible?

Reply:
I'm not sorry 😞